xclim/indices/_agro.py

```diff
@@ -589,6 +595,225 @@ def water_budget(
     return out
 
 
+@declare_units(
+    pr="[precipitation]",
+    pr_cal="[precipitation]",
+)
+def standardized_precipitation_index(
+    pr: xarray.DataArray,
+    pr_cal: xarray.DataArray,
+    freq: str = "MS",
+    window: int = 1,
+    dist: str = "gamma",
+    method: str = "ML",
+) -> xarray.DataArray:
+    r"""Standardized Precipitation Index (SPI).
+
+    Parameters
+    ----------
+    pr : xarray.DataArray
+      Daily precipitation.
+    pr_cal : xarray.DataArray
+      Daily precipitation used for calibration. Usually this is a temporal subset of `pr` over some reference period.
+    freq : str
+      Resampling frequency. A monthly or daily frequency is expected.
+    window : int
+      Averaging window length relative to the resampling frequency. For example, if `freq="MS"`, i.e. a monthly resampling, the window
+      is an integer number of months.
+    dist : str
+      Name of the univariate distribution, such as `beta`, `expon`, `genextreme`, `gamma`, `gumbel_r`, `lognorm`, `norm`
+      (see :py:mod:`scipy.stats`).
+    method : str
+      Name of the fitting method, such as `ML` (maximum likelihood), `APP` (approximate). The approximate method
+      simply uses the parameters output of `_fit_start`.
```

Comment:
```suggestion
      uses a deterministic function that doesn't involve any optimization.
```
Not sure how to phrase it, but I don't think most users will be familiar with `_fit_start`.